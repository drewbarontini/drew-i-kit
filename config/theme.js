const theme = {
  borders: {
    radius: {
      base: '3px',
    },
  },
  colors: {
    bg: '#fff',
    fg: '#272727',
    primary: '#3576a7',
    subdued: '#cbcbcb',
    error: '#fe5f55',
    success: '#8acb88',
    warning: '#ffc527',
  },
  fonts: {
    family: {
      base:
        "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
      heading:
        "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
    },
    lineHeight: {
      base: '1.5',
      heading: '1.2',
    },
    sizes: {
      base: '14px',
      xxs: '8px',
      xs: '10px',
      s: '12px',
      l: '16px',
      xl: '24px',
      h1: '26px',
    },
    weights: {
      light: 200,
      normal: 400,
      semiBold: 600,
      bold: 800,
    },
  },
  spacing: {
    base: '16px',
    xs: '5px',
    s: '10px',
    m: '12px',
    l: '32px',
    xl: '60px',
  },
};

export default theme;
