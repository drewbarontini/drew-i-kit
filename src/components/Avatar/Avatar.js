// DEPENDENCY: <Image />

import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { ifProp } from 'styled-tools';

import { theme, palette } from '../../lib/props';
import { responsive } from '../../lib/responsive';

import Image from '../Image';

const Avatar = styled(Image)`
  ${ifProp(
    'raised',
    css`
      border: 5px solid ${palette('white')};
      box-shadow: ${theme('shadows.base')};
    `
  )}

  ${ifProp(
    'rounded',
    css`
      border-radius: 100%;
    `
  )}

  ${responsive};
`;

Avatar.propTypes = {
  raised: PropTypes.string,
  rounded: PropTypes.string,
};

Avatar.displayName = 'Avatar';

export default Avatar;
