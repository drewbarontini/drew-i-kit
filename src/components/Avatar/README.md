# Avatar

## Usage

```jsx
import { Avatar } from 'drew-i-kit';

const standardProps = {
  src: 'https://source.unsplash.com/random/200x200',
  alt: 'Unsplash Image',
  width: '100',
  height: '100',
};

<Avatar {...standardProps} />
```

```jsx
import { Avatar } from 'drew-i-kit';

<Avatar rounded {...standardProps} />
<Avatar raised {...standardProps} />
<Avatar rounded raised {...standardProps} />
```

## Props

- `alt` (**string**)
- `height` (**string**, optional)
- `raised` (**Boolean**, optional)
- `rounded` (**Boolean**, optional)
- `src` (**string**)
- `width` (**string**, optional)