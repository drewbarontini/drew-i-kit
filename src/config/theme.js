import colors from './colors';
import fonts from './fonts';
import icons from './icons';

const theme = {
  borders: {
    sizes: {
      base: '1px',
      l: '2px',
    },
    radius: {
      base: '3px',
    },
  },
  breakpoints: {
    sizes: {
      s: 380,
      m: 768,
      l: 900,
      xl: 1100,
    },
    query: 'max-width',
    unit: 'px',
  },
  colors: {
    ...colors,
  },
  fonts: {
    ...fonts,
  },
  icons: {
    ...icons,
  },
  shadows: {
    base: '0 3px 3px rgba(0, 0, 0, 0.25);',
  },
  sizing: {
    base: '900px',
    s: '400px',
    l: '1200px',
  },
  spacing: {
    base: '20px',
    xs: '5px',
    s: '10px',
    l: '40px',
    xl: '60px',
  },
  transition: {
    base: '0.3s ease-in-out',
  },
};

export default theme;
