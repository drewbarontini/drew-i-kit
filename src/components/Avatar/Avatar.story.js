import React from 'react';
import { storiesOf } from '@storybook/react';

import Avatar from '.';

const standardProps = {
  src: 'https://source.unsplash.com/random/200x200',
  alt: 'Unsplash Image',
  width: '100',
  height: '100',
};

storiesOf('Avatar', module)
  .add('default', () => <Avatar {...standardProps} />)
  .add('rounded', () => <Avatar rounded {...standardProps} />)
  .add('raised', () => <Avatar raised {...standardProps} />)
  .add('rounded and raised', () => (
    <Avatar rounded raised {...standardProps} />
  ));
