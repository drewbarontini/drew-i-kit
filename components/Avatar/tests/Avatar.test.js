import React from 'react';
import { render } from 'react-testing-library';
import 'jest-styled-components';

import theme from '../../../config/theme';

import Avatar from '../';

describe('<Avatar />', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <Avatar src="IMG_SRC" alt="Alt Text" width="50" height="50" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
