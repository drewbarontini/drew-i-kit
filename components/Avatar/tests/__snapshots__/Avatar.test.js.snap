// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> should match snapshot 1`] = `
.c0 {
  height: 50px;
  width: 50px;
}

<div
  class="c0"
  height="50"
  width="50"
/>
`;
